deno_dist/validator/validator.ts

```diff
@@ -61,7 +69,20 @@ export const validator = <
         }
         break
       case 'form':
-        value = await c.req.parseBody()
+        await (async () => {
+          const contentType = c.req.header('Content-Type')
+          if (contentType) {
+            const arrayBuffer = c.req.bodyCache.arrayBuffer ?? (await c.req.raw.arrayBuffer())
+            const formData = arrayBufferToFormData(arrayBuffer, contentType)
+            const form: BodyData = {}
+            formData.forEach((value, key) => {
+              form[key] = value
+            })
+            value = form
+            c.req.bodyCache.formData = formData
+            c.req.bodyCache.arrayBuffer = arrayBuffer
+          }
+        })()
```

Comment:
I think we dont need to create a new `async function` in here, we can just change the code into something like this

```ts
   case 'form': {
      value = await c.req.parseBody();
      const contentType = c.req.header('Content-Type');

      // Just skip if no content type

      if (!contentType) break;
      const arrayBuffer =
        c.req.bodyCache.arrayBuffer ?? (await c.req.raw.arrayBuffer());
      const formData = arrayBufferToFormData(arrayBuffer, contentType);
      const form: BodyData = {};
      formData.forEach((value, key) => {
        form[key] = value;
      });
      value = form;
      c.req.bodyCache.formData = formData;
      c.req.bodyCache.arrayBuffer = arrayBuffer;

      break;
    }
```